app/questionnaire/placeholder_renderer.py

```diff
@@ -149,5 +166,56 @@ def render(
                 placeholder_parser=placeholder_parser,
             )
             set_pointer(data_to_render_mutable, pointer, rendered_text)
-
         return data_to_render_mutable
+
+    def resolve_dynamic_answers(
+        self, data_to_render_mutable: dict, list_items: list[str]
+    ) -> None:
+        dynamic_answers = data_to_render_mutable["dynamic_answers"]
+        if dynamic_answers["values"]["source"] == "list":
```

Comment:
Do we need to handle placeholder resolution for dynamic answers sourced from "answers" instead of "list"? I can see they're piped in as answer options for dynamic answer questions but could there be a scenario in which these "answers" sourced dynamic answer options are used for question text and thus need placeholder resolution like is being handled here?

Reply:
+ function driven dynamic answers but I think they count as "answers" sourced dynamic answer options

Reply:
Kyle not at the moment, we abandoned the idea of resolving based on "answers" (for now). There is a comment on the card: "A decision was made to only support dynamic answers based on a list collector". Same with function driven.

Reply:
Ah yep I see that comment now, no worries at all my bad